src/javascript/app_2/Stores/Modules/Contract/contract-store.js

```diff
@@ -100,14 +99,17 @@ export default class ContractStore extends BaseStore {
         this.forgetProposalOpenContract();
         this.contract_id       = null;
         this.contract_info     = {};
+        this.forget_id         = null;
         this.digits_info       = {};
         this.sell_info         = {};
         this.is_sell_requested = false;
         this.chart_config      = {};
+        this.is_left_epoch_set = false;
 
         destroyChartTickMarkers();
         this.smart_chart.removeBarriers();
         this.smart_chart.removeMarkers();
+        this.smart_chart.resetScrollZoom();
         this.smart_chart.setContractMode(false);
```

Comment:
Could these all be put in a `this.smart_chart.cleanupContractChartView()` or similar?